src/screens/DeviceStack/KeystoneStack/keystone-accounts.tsx

```diff
@@ -0,0 +1,179 @@
+  >([]);
+
+  const generator = useRef<ReturnType<typeof getWalletsFromProvider> | null>(
+    null,
+  );
+
+  const goBack = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  const createWalletGenerator = () => {
+    if (provider.current !== null) {
+      generator.current = getWalletsFromProvider(
+        provider.current,
+        WalletType.keystone,
+      );
+    }
+  };
+
+  const loadMore = useCallback(
+    async (reset: boolean = false) => {
+      if (loading) {
+        return;
+      }
+      if (reset || generator.current === null) {
+        createWalletGenerator();
+      }
+      setLoading(true);
+      let index = 0;
+      let result = reset ? [] : addresses;
+      while (index < PAGE_SIZE) {
+        if (generator.current) {
+          const item = (await generator.current.next()).value;
+          result.push(item);
+        }
+        index += 1;
+      }
+      const wallets = result.map(item => item.address);
+      const balances = await Indexer.instance.updates(
+        wallets.map(AddressUtils.toHaqq),
+        new Date(0),
+      );
+      const resultWithBalances = result.map(item => ({
+        ...item,
+        balance: new Balance(
+          balances.total[AddressUtils.toHaqq(item.address)] || item.balance,
+        ),
+      }));
+      setAddresses(resultWithBalances);
+      setLoading(false);
+    },
+    [addresses, loading],
+  );
+
+  useEffectAsync(async () => {
+    setLoading(true);
+    try {
+      await loadMore();
+    } catch (error) {
+      Logger.captureException(error, 'chooseAccount');
+      switch (error) {
+        case 'wallet_already_exists':
+          showModal(ModalType.errorAccountAdded);
+          goBack();
+          break;
+        default:
+          if (error instanceof Error) {
+            Logger.log('error.message', error.message);
+            showModal(ModalType.errorCreateAccount);
+            goBack();
+          }
+      }
+    } finally {
+      setLoading(false);
+    }
+  }, []);
+
+  const data = useMemo(
+    (): ChooseAccountItem[] =>
+      addresses.map(item => ({
+        ...item,
+        hdPath: item.path,
+        // publicKey: '',
+        exists:
+          walletsToCreate.find(wallet => wallet.address === item.address)
+            ?.exists ||
+          item.exists ||
+          false,
+      })),
+    [addresses, walletsToCreate],
+  );
+
+  const onItemPress = useCallback(
+    (item: ChooseAccountItem) => {
+      const newAddresses = addresses.map(address => {
+        if (item.address === address.address) {
+          return {
+            ...address,
+            exists: !item.exists,
+          };
+        }
+        return address;
+      });
+
+      setAddresses([...newAddresses]);
+      updateWalletsToCreate(newAddresses.filter(address => !!address.exists));
+    },
+    [addresses],
+  );
+
+  const onAdd = useCallback(async () => {
+    walletsToCreate.forEach(item => {
+      Wallet.create(item.name, item);
```

Comment:
Please add support for `isImported` field in `.create` method.
```suggestion
      Wallet.create(item.name, {...item, isImported: true});
```

Or you can update created wallet:
```
Wallet.update(item.name, {isImported: true})
```